homeassistant/components/image_processing/__init__.py

```diff
@@ -0,0 +1,253 @@
+"""
+Provides functionality to interact with image processing services.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/image_processing/
+"""
+import asyncio
+from datetime import timedelta
+import logging
+import os
+
+import aiohttp
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_ENTITY_PICTURE, CONF_TIMEOUT,
+    CONF_NAME, CONF_ENTITY_ID, STATE_UNKNOWN)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.util.async import run_callback_threadsafe
+
+DOMAIN = 'image_processing'
+DEPENDENCIES = ['camera']
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=10)
+
+PROCESSING_CLASSES = [
+    None,            # Generic
+    'alpr',          # Processing licences plates
+    'face',          # Processing biometric face data
+]
+
+SERVICE_SCAN = 'scan'
+
+EVENT_FOUND_PLATE = 'found_plate'
+
+ATTR_PLATE = 'plate'
+ATTR_CONFIDENCE = 'confidence'
+
+CONF_SOURCE = 'source'
+CONF_CONFIDENCE = 'confidence'
+
+DEFAULT_TIMEOUT = 10
+DEFAULT_CONFIDENCE = 80
```

Comment:
Yeah, all image processing api have confidence since a computer can only say how possible that a case is but not exact. So I add this to base, but I can also remove it to all platforms..